sdk/src/main/scala/io/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -249,7 +258,10 @@ private object WebSocketAccountServerEndpoint extends SparkzLogging {
   }
 
   def onVaultChanged(): Unit = {
-    walletAddresses = webSocketAccountChannelImpl.getWalletAddresses
+    webSocketAccountChannelImpl.getWalletAddresses match {
+      case Success(addresses) => walletAddresses = addresses
+      case Failure(exception) => log.debug("Websocket failed to update walletAddresses "+exception.getMessage)
```

Comment:
Currently, in case wallet has been changed (new secret added), but this method failed because of a timeout -> we will never leave `walletAddresses` as before.
Do we want instead to try again until we succeed?

Reply:
I suppose that if we are not able to access the wallet, adding a retry would just decrease more tha node performance. We may consider to add some sort of queue that keeps track of the failed requests and try to re-executed them in a second moment.